DatadogInternal/Sources/NetworkInstrumentation/NetworkInstrumentationFeature.swift

```diff
@@ -29,7 +29,7 @@ internal final class NetworkInstrumentationFeature: DatadogFeature {
     )
 
     /// A no-op message bus receiver.
-    internal let messageReceiver: FeatureMessageReceiver = NOPFeatureMessageReceiver()
+    internal let messageReceiver: FeatureMessageReceiver? = NOPFeatureMessageReceiver()
```

Comment:
suggestion/  Now we have two ways of achieving the same: either setting `messageReceiver` to `nil` or to `NOP*` will have the same result. This would be error-prone on the long term as `!= nil` check won't reliably determine if the feature can receive messages or not. I'd suggest allowing either way, not both.

Reply:
I agree! I think the `NOP*` pattern can stay IMHO, but either way is fine.
Additionally, I think the `performanceOverride` is actually mis-placed and should be in the `DatadogRemoteFeature`.
Additionally 2, the `DatadogFeatureConfiguration` defined above is a leftover from the migration and no longer in use.